src/integration/sink/sink.go

```diff
@@ -15,13 +15,13 @@ const (
 	DefaultAgentForwarderPath = "/v1/data"
 )
 
-// Doer is the interface that httpSink client should satisfy.
+// Doer is the interface that HTTPSink client should satisfy.
 type Doer interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
-// httpSink holds the configuration of the HTTP sink used by the integration.
-type httpSink struct {
+// HTTPSink holds the configuration of the HTTP sink used by the integration.
+type HTTPSink struct {
```

Comment:
why now it is exported? Should we return this instead of the interface from .New()?

Reply:
I think this was due to some refactor I did not fully complete. I prefer exported types so I'll change the return type for `New` :)